baybe/objective.py

```diff
@@ -110,8 +110,8 @@ def transform(self, data: pd.DataFrame) -> pd.DataFrame:
                 contain more columns.
 
         Returns:
-            A dataframe with the targets in computational representation. Columns will
-            be as in the input (except when objective mode is ``DESIRABILITY``).
+            A new dataframe with the targets in computational representation. Columns
```

Comment:
Yes, this is to make it explicit what happens in the back. Otherwise, you simply wouldn't know from the docstring if the manipulation happens inplace or not. If you check the history, you'll find a commit "Cleanly handle data copies" that specifically cleans up this stuff. I noticed that there was no clear structure what is copied and what not, and from the existing docstring I could also not conclude it. So I went into the code, made copies only where really necessary, and changed the docstring accordingly. I'd be happy if you could double check that commit and let me know if you agree.

Reply:
Ok, so to get this clear: You wanted to make clear that the returned object will *always* be a copy, even if there is no need for transforming/changing the input data at all, right? In that case, adding the "new" makes sense to emphasize this. Also had a look at the commit, the changed operations look good to me. So feel free to resolve :)

Reply:
Not sure how I should interpret your _"even if there is no need for transforming/changing the input data at all"_?? Could you explain? But apart from that, yes!

Reply:
Ok, glad that we agree. But just FYI: I don't think there can ever be a difference between "new" and "completely new" for dataframes. Either it's a new object, or not.

Reply:
That *completely* was just added to make *completely* clear what I mean 😉